Software/bsw/static/Mcal/CAN/src/Can.c

```diff
@@ -1689,51 +1720,49 @@ return  Return_type;
 Std_ReturnType Can_GetControllerMode(uint8 Controller,Can_ControllerStateType* ControllerModePtr) 
 {
 	Std_ReturnType Loc_Can_GetControllerMode_Ret = E_OK;
-
-#if(CanDevErrorDetect == STD_ON)
-	if (CAN_CS_UNINIT==ControllerState[Controller])
-	{
-		Det_ReportError(CAN_MODULE_ID,CAN_INSTANCE_ID,Can_GetControllerMode_Id,CAN_E_UNINIT);
-		Loc_Can_GetControllerMode_Ret = E_NOT_OK;
-	}
-	else
-	{
-	    /* Do Nothing */
-	}
-
-#endif
-
-	if ( NUM_OF_CAN_CONTROLLERS <= Controller )
-	{
-		Det_ReportError(CAN_MODULE_ID, CAN_INSTANCE_ID,Can_GetControllerMode_Id, CAN_E_PARAM_CONTROLLER);
-		Loc_Can_GetControllerMode_Ret = E_NOT_OK;
-	} 
-	else 
+    /*  [SWS_Can_91016] If development error detection for the Can module is enabled:
+     *  The function Can_GetControllerMode shall raise the error CAN_E_UNINIT and
+     *  return E_NOT_OK if the driver is not yet initialized.
+     *  ( SRS_BSW_00406,SRS_BSW_00416)
+     */
+	if(ModuleState == CAN_UNINIT)
 	{
-	    /* Do Nothing */
+        #if(CanDevErrorDetect == STD_ON)
+	        Det_ReportError(CAN_MODULE_ID, CAN_INSTANCE_ID, Can_GetControllerErrorState_Id, CAN_E_UNINIT);
+	        Loc_Can_GetControllerMode_Ret = E_NOT_OK;
+        #endif
 	}
-	if ( NULL_PTR == ControllerModePtr)
-	{
-		Det_ReportError(CAN_MODULE_ID, CAN_INSTANCE_ID,Can_GetControllerMode_Id, CAN_E_PARAM_POINTER);
-		Loc_Can_GetControllerMode_Ret = E_NOT_OK;
-	} 
-	else
+	/*[SWS_Can_91017] If parameter Controller of Can_GetControllerMode() has an invalid value,
+	 *  the CanDrv shall report development error code CAN_E_PARAM_CONTROLLER to the Det_ReportError
+	 *  service of the DET.(SRS_BSW_00323)
+	 */
+	else if(NUM_OF_CAN_CONTROLLERS <= Controller)
 	{
-	    /* Do Nothing */
+        #if(CanDevErrorDetect == STD_ON)
+	        Det_ReportError(CAN_MODULE_ID, CAN_INSTANCE_ID,Can_GetControllerMode_Id, CAN_E_PARAM_CONTROLLER);
+	        Loc_Can_GetControllerMode_Ret = E_NOT_OK;
+        #endif
 	}
-
-	if ( E_NOT_OK != Loc_Can_GetControllerMode_Ret)
+	/*[SWS_Can_91018] If parameter ControllerModePtr of Can_GetControllerMode() has an null pointer,
+	 *  the CanDrv shall report development error code CAN_E_PARAM_POINTER to the Det_ReportError
+	 *   service of the DET. (SRS_BSW_00323)
+	 */
+	else if(NULL_PTR == ControllerModePtr)
 	{
-
-		*ControllerModePtr = ControllerState[Controller];
+        #if(CanDevErrorDetect == STD_ON)
+	        Det_ReportError(CAN_MODULE_ID, CAN_INSTANCE_ID,Can_GetControllerMode_Id, CAN_E_PARAM_POINTER);
+	        Loc_Can_GetControllerMode_Ret = E_NOT_OK;
+        #endif
 	}
+	/*
+	 * [SWS_Can_91015] The service Can_GetControllerMode shall return the mode
+	 * of the requested CAN controller.
+	 */
 	else
 	{
-	    /* Do Nothing */
+	    *ControllerModePtr = ControllerState[Controller];
```

Comment:
I think this should not relate to the DEV checks, it can be separately without else if all checks successfully passed

Reply:
No , because the function only returns the state if no error found as I understood from this
![image](https://user-images.githubusercontent.com/32249454/70516059-f6c13180-1b3e-11ea-8d43-621d1219858d.png)